hack/generator/pkg/astmodel/type_visitor.go

```diff
@@ -128,23 +128,29 @@ func IdentityVisitOfPrimitiveType(_ *TypeVisitor, it *PrimitiveType, _ interface
 	return it, nil
 }
 
-func identityVisitObjectTypePerPropertyContext(_ *PropertyDefinition, ctx interface{}) interface{} {
-	return ctx
+func identityVisitObjectTypePerPropertyContext(_ *ObjectType, _ *PropertyDefinition, ctx interface{}) (interface{}, error) {
+	return ctx, nil
```

Comment:
I did. The `UnrollObjectTypeProperty` function adheres to this signature, and has a possible error return if/when `AddAllowDuplicates` returns an `err`.

The whole reasoning for this is so that we can have a "per-property handler" for the visitor rather than a per-object type handler. Before this there were a good number of places where we were duplicating 95% of the default object visitor code because there wasn't a per-property hook.